### PR TITLE
e2e: add test for dynamic balloons configuration update

### PR DESCRIPTION
**Pull request by askervin**
- Add new test that changes balloons and CPU configurations using a
  config map.
- Refactor verify-metrics functions to a common source file.

### CODE REVIEW

**Comment by askervin** [comment]
Needs balloons configuration update patch from PR #830.

File: test/e2e/policies.test-suite/balloons/n4c16/test06-update-configmap/code.var.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    terminate cri-resmgr
+    terminate cri-resmgr-agent
+    vm-command "cri-resmgr -reset-policy; cri-resmgr -reset-config"
+}
+
+apply-configmap() {
+    vm-put-file $(instantiate balloons-configmap.yaml) balloons-configmap.yaml
+    vm-command "cat balloons-configmap.yaml"
+    kubectl apply -f balloons-configmap.yaml
+}
```
Comment (klihub) [inline test/e2e/policies.test-suite/balloons/n4c16/test06-update-configmap/code.var.sh:19]: Hmm... shouldn't it be something like:

```
cleanup() {
    vm-command "kubectl delete pods --all --now; \
        kubectl delete pods -n $testns --all --now; \
        kubectl delete pods -n btype1ns0 --all --now; \
        kubectl delete namespace $testns || :; \
        kubectl delete namespace btype1ns0 || :"
    terminate cri-resmgr
    terminate cri-resmgr-agent
    vm-command "cri-resmgr -reset-policy; cri-resmgr -reset-config"
}

apply-configmap() {
    vm-put-file $(instantiate balloons-configmap.yaml) balloons-configmap.yaml
    vm-command "cat balloons-configmap.yaml"
    vm-command "kubectl apply -f balloons-configmap.yaml"
}
```

**Review by klihub — CHANGES_REQUESTED** [review]

File: test/e2e/policies.test-suite/balloons/n4c16/test06-update-configmap/code.var.sh [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+verify-metrics-has-line '"btype2\[0\]".*pod2:pod2c0'
+
+# Change only CPU classes, no reassigning.
+verify-metrics-has-line 'btype0\[0\].*pod0:pod0c0.*cpu_class="classA"'
+verify-metrics-has-line 'btype2\[0\].*pod1:pod1c0.*cpu_class="classC"'
+verify-metrics-has-line 'btype2\[0\].*pod2:pod2c0.*cpu_class="classC"'
+BTYPE0_CPUCLASS="classC" BTYPE1_SKIP=1 BTYPE2_CPUCLASS="classB" BTYPE2_NAMESPACE0='"*"'  apply-configmap
+verify-metrics-has-line 'btype0\[0\].*pod0:pod0c0.*cpu_class="classC"'
+verify-metrics-has-line 'btype2\[0\].*pod1:pod1c0.*cpu_class="classB"'
+verify-metrics-has-line 'btype2\[0\].*pod2:pod2c0.*cpu_class="classB"'
```
Comment (klihub) [inline test/e2e/policies.test-suite/balloons/n4c16/test06-update-configmap/code.var.sh:72]: And do a final
```
cleanup
```

here at the end too, please.

File: test/e2e/policies.test-suite/balloons/n4c16/test06-update-configmap/code.var.sh [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        kubectl delete namespace btype1ns0 || :"
+    terminate cri-resmgr
+    terminate cri-resmgr-agent
+    vm-command "cri-resmgr -reset-policy; cri-resmgr -reset-config"
+}
+
+apply-configmap() {
+    vm-put-file $(instantiate balloons-configmap.yaml) balloons-configmap.yaml
+    vm-command "cat balloons-configmap.yaml"
+    kubectl apply -f balloons-configmap.yaml
```
Comment (klihub) [inline test/e2e/policies.test-suite/balloons/n4c16/test06-update-configmap/code.var.sh:20]: Hmm... isn't this still wrong ? Shouldn't it be `vm-command "kubectl apply -f balloons-configmap.yaml"`

Comment (klihub) [reply test/e2e/policies.test-suite/balloons/n4c16/test06-update-configmap/code.var.sh:20]: Okay, I see. We have a `kubectl()` wrapper shell function defined in `run.sh`. I have to say, the way how we use both that and also `vm-command "kubectl..."`'s is totally super-confusing... We should at least try to be more consistent with these.

**Review by klihub — APPROVED** [review]
LGTM

**Pull request merged**